lookupy/lookupy.py

```diff
@@ -314,11 +314,14 @@ def iff(precond, val, f):
 
 
 def guard_type(classinfo, val):
-    if not isinstance(val, classinfo):
+    if isinstance(classinfo, tuple) and not isinstance(val, classinfo[0]) and not isinstance(val, classinfo[1]):
```

Comment:
Here only 2-tuples will be supported. You can allow checking for any no. of types by using `any` or `all` functions.